apps/simple-trading-app-e2e/src/integration/market-trade.test.ts

```diff
@@ -0,0 +1,67 @@
+import { aliasQuery } from '@vegaprotocol/cypress';
+import { generateSimpleMarkets } from '../support/mocks/generate-markets';
+import { generateDealTicket } from '../support/mocks/generate-deal-ticket';
+
+describe('Market trade', () => {
+  let markets;
+  beforeEach(() => {
+    cy.mockGQL((req) => {
+      aliasQuery(req, 'SimpleMarkets', generateSimpleMarkets());
+      aliasQuery(req, 'DealTicketQuery', generateDealTicket());
+    });
+    cy.visit('/markets');
+    cy.wait('@SimpleMarkets').then((response) => {
+      if (response.response?.body?.data?.markets?.length) {
+        markets = response.response?.body?.data?.markets;
+      }
+    });
+  });
+  it('side selector should work well', () => {
+    if (markets) {
```

Comment:
Question (blocking?): Because 'markets' could be an empty array, will line 21 fail? If so, better if this was `if (markets.length > 0)`. Same on line 37.

Reply:
if (markets.length)

Reply:
I expect here `markets` can be undefined or not empty array. This is what provides line 15 above:

```
if (response.response?.body?.data?.markets?.length) {
        markets = response.response?.body?.data?.markets;
```
So I'm gonna put there something like:

`if (markets?.length) {`

should cover all cases and be more readable